packages/design-system/src/scalars/components/number-field/numberFieldValidations.ts

```diff
@@ -0,0 +1,45 @@
+import { NumberFieldProps } from "./number-field";
+
+export const validatePositive =
+  (props: NumberFieldProps) => (value: string) => {
+    return props.allowNegative
+      ? true
+      : Number(value) > 0
+        ? true
+        : "Value must be a positive value";
+  };
+
+export const validateIsBigInt =
+  (props: NumberFieldProps) => (value: string) => {
+    const isLargeNumber = Math.abs(Number(value)) > Number.MAX_SAFE_INTEGER;
+    return isLargeNumber && !props.isBigInt
+      ? "Value is too large for standard integer, set isBigInt to true"
```

Comment:
You should omit this part: "set isBigInt to true". This is a message to be presented to the users no developers, and users can not change the code

Reply:
You are right, done